### DISCUSSION TITLE
[Feature] (Android) Support for identifying burst photos (continuous shooting)

### DISCUSSION THREAD

**Discussion by Rydberg95** [immich-app/immich#1885] [Feature Request]
### The feature

There is a [discussion already for iOS](https://github.com/immich-app/immich/discussions/1770), but since Android isn't mentioned there, I thought I would open a separate discussion for that.

It would be great if support could be added to group/handle photos that were captured in continuous shooting (burst) mode. Capturing photos at 20 fps quickly bloats the gallery... :)

There is support for this in Google Photos, so a lot of inspiration could be found there. See screenshots below for how the UI looks there.

### Identifying burst files, naming scheme
Now, the tricky thing when handling burst sequences would be how photos are identified as belonging to a burst sequence, I think there might have to be different solutions to this since I've seen different structures in place depending on phone model or app. However, I will be taking my case (Sony Xperia 1 IV) as an example, which I know I've seen with previous models as well. Each burst sequence is placed in a separate subfolder (which is worth mentioning, but imo not relevant). The naming scheme is, with dynamic parts inside curly brackets:

DSC_{ BURST_IMAGE_NUMBER }_BURST{ TIMESTAMP }.JPG

Additional information on dynamic parts:
BURST_IMAGE_NUMBER: The sequence number for that image. First image gets number 0000, second image gets image 0001.
TIMESTAMP: A timestamp that **is the same for the entire sequence**. Formatted as _YYYYMMDDhhmmssfff_. For an example, see the last screenshot below.

### Identifying burst files, EXIF-data
Seems there is EXIF-data that might be giving away if the image is part of a burst sequence. The relevant entries I've found are below.

**First image in burst sequence:**
Burst ID                        : 18a88b41-f774-4e63-8e44-e275b2f58d44
Burst Primary                : 1

**Second (and later) image in burst sequence:**
Burst ID                        : 18a88b41-f774-4e63-8e44-e275b2f58d44

(Obviously) "Burst ID" seems to be the unique identifier for the burst sequence, and is the same for all images in the sequence. "Burst Primary" seems to be a flag for just the first image. The "Burst Primary" entry is only present for the first image. There does not seem to be any info in the EXIF-data for what number place that image has in the sequence. I guess that could be ordered based on creation time, alternatively on the filename mentioned above.

### Google Photos, Android:
![signal-2023-02-26-195714_002](https://user-images.githubusercontent.com/24879708/221430973-50fd98dd-d6f2-46b5-9902-a8390a3fd2a2.png)

### Google Photos, Web:
![bild](https://user-images.githubusercontent.com/24879708/221431027-dba096aa-2a2d-4e0b-b875-8cf5223de25e.png)

### Filename structure:
![bild](https://user-images.githubusercontent.com/24879708/221431366-31ae5769-3c1f-4da4-be8f-b25194b20e6d.png)

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by soubiliz** [comment]
This is a screenshot of aves gallery that recognize 3 burst shot pattern, maybe it's worth checking out their implementation 
![Screenshot_20240421_095651_Aves](https://github.com/immich-app/immich/assets/19488257/30cf2d19-76e3-4c3d-a5c0-26b1943878a5)

Edit:
They seam to only look at file names :
https://github.com/deckerst/aves/blob/569aef5f9f1e1b3d39df5df061a7ace4b6f891b1/lib/ref/bursts.dart#L4
